network-api/networkapi/wagtailpages/pagemodels/publications/article.py

```diff
@@ -71,5 +68,54 @@ class ArticlePage(FoundationMetadataPageMixin, Page):
             InlinePanel("authors", label="Author", min_num=0)
         ], heading="Author(s)"),
         StreamFieldPanel('body'),
+        InlinePanel("footnotes", label="Footnotes"),
         FieldPanel('sidebar_summary_title'),
     ]
+
+    @property
+    def next_page(self) -> Union[Page, None]:
+        # Try to get the next sibling page
+        next_page = self.get_next_sibling()
+        if next_page:
+            return next_page
+        # No next sibling page exists,
+        # Get the parent page and check if it's an ArticlePage
+        parent_sibling = self.get_parent().get_next_sibling()
+        if parent_sibling:
+            # Check if parenet page is an ArticlePage
+            if isinstance(parent_sibling.specific, ArticlePage):
+                return parent_sibling
+            # Parent page was not an ArticlePage, return the first
+            # live child, or None
+            return parent_sibling.get_children().live().first()
+
+    @property
+    def prev_page(self) -> Union[Page, None]:
+        # Try to get the prev sibling page
+        prev_page = self.get_prev_sibling()
+        if prev_page:
+            return prev_page
+        # No prev sibling page exists,
+        # Get the parent page and check if it's an ArticlePage
+        parent_sibling = self.get_parent().get_prev_sibling()
+        if parent_sibling:
+            # Check if parenet page is an ArticlePage
+            if isinstance(parent_sibling.specific, ArticlePage):
+                return parent_sibling
+            # Parent page was not an ArticlePage, return the first
+            # live child, or None
+            return parent_sibling.get_children().live().last()
+
+    def get_titles(self):
```

Comment:
i'd be interested to see what affect getting these titles this way has on performance *nerd glasses emoji*

Reply:
Should be very little. All stream data is kept in a single field as a dictionary, then we loop through it to find "content" streamblocks, and use BeautifulSoup to pull out the `h2` elements. There are no additional queries, the biggest thing is BS4 parsing the HTML (Or rather, the "wagtail" HTML) and that should be quite minimal as it's not ever saved as deeply nested HTML. That said, I'd be interested to see the performance on this as well. That could be a spike ticket though.